pkg/gen/input/workflows/internal/file/gitleaks.go

```diff
@@ -0,0 +1,38 @@
+package file
+
+import (
+	"github.com/giantswarm/devctl/pkg/gen/input"
+	"github.com/giantswarm/devctl/pkg/gen/input/workflows/internal/params"
+)
+
+func NewGitleaksInput(p params.Params) input.Input {
+	i := input.Input{
+		Path:         params.RegenerableFileName(p, "gitleaks.yaml"),
+		TemplateBody: gitleaksTemplate,
+		TemplateDelims: input.InputTemplateDelims{
+			Left:  "{{{{",
+			Right: "}}}}",
+		},
+	}
+
+	return i
+}
+
+var gitleaksTemplate = `# DO NOT EDIT. Generated with:
+#
+#    devctl gen workflows
+#
+name: gitleaks
+
+on: [push,pull_request]
+
+jobs:
+  gitleaks:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: '0'
```

Comment:
It has to pull the whole history because the action has to be able to find the most recent common ancestor with the target branch. It doesn't scan the whole thing

Reply:
I'm more concerned about time consumed to pull whole repo on every commit putting performance of the tool aside.

Reply:
It takes ~6 seconds on `kvm-operator`, 8 on `aws-operator`, and 2 on `release-operator`. But in any case I don't think there's a workaround since we don't know how many commits back the last ancestor will be

Reply:
Okay. We don't have a changelog :P

Reply:
Haha yep, added an entry